### PR TITLE
[ConstraintSystem] Don't increase the score for unapplied function references if the expression is an argument to `#selector`.

### PR DESCRIPTION
**Pull request by hborla**
For `#selector` arguments, functions and properties are syntactically distinct with the getter/setter label, so the solver should not unconditionally prefer properties to unapplied functions. 

Note that a better fix for this is to port over the `#selector` diagnostics from CSApply to constraint fixes, and not attempt invalid disjunction choices based on the selector kind on the valid code path.

Resolves: rdar://83779014

### CODE REVIEW

File: lib/Sema/ConstraintSystem.cpp [hunk 1]
```diff
@@ -2916,7 +2916,9 @@ void ConstraintSystem::resolveOverload(ConstraintLocator *locator,
     }
 
     if (isa<AbstractFunctionDecl>(decl) || isa<TypeDecl>(decl)) {
-      if (choice.getFunctionRefKind() == FunctionRefKind::Unapplied) {
+      auto *anchor = locator->getAnchor();
```
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:2919]: Anchors are `ASTNode`s now :)

**Review by xedin — COMMENTED** [review]
Might be worth adding what you have in “note” section of the description as a TODO where the score is increased.

File: lib/Sema/ConstraintSystem.cpp (hunk 1 above)
Comment (hborla) [reply lib/Sema/ConstraintSystem.cpp]: 🤦🏻‍♀️ fixed!

**Comment by hborla** [comment]
@swift-ci please test source compatibility

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please test Windows platform

**Pull request merged**